packages/web/src/lib/ai-autocomplete/language_server.proto

```diff
@@ -0,0 +1,172 @@
+// NOTE: to generate a new set of json definitions, run:
+// npm run generate-codeium-proto-json
+
+// Copyright Exafunction, Inc.
+
+syntax = "proto3";
+
+package exa.language_server_pb;
+
+import "codeium_common.proto";
```

Comment:
Here's the underlying protobuf definitions that are used to generate the codeium autocomplete request. These came from here: https://github.com/Exafunction/codeium-react-code-editor/blob/main/exa/language_server_pb/language_server.proto.

Note that to avoid including this raw text in the bundle that is sent to the client, I converted the protobuf file to json descriptiors ([more about these here](https://github.com/protobufjs/protobuf.js?tab=readme-ov-file#using-json-descriptors)). These are a fair bit smaller and work just as well.

Another option I considered but ultimately decided against given the complexity of introducing was an alternate protobuf library that does code generation at build time, like [`buf`](https://buf.build/docs/introduction). If protobufs were going to be used everywhere and definitions were going to be changing often, I think it would be a lot of sense, but given this is only used here that seemed like overkill.

Reply:
@1egoman are we okay with the copyright on this?